library/std/src/error.rs

```diff
@@ -1547,16 +1539,18 @@ impl<E> Report<E> {
 
 impl<E> Report<E>
 where
+    // FIXME(yaahc): Remove this 'static bound once error trait has been moved
+    // into core and a blanket impl of `Provider` has been added for `Error`.
     E: Error,
 {
     fn backtrace(&self) -> Option<&Backtrace> {
         // have to grab the backtrace on the first error directly since that error may not be
         // 'static
-        let backtrace = self.error.backtrace();
+        let backtrace = (&self.error as &dyn Error).request_ref();
```

Comment:
I'm not happy with the ergonomics here, though I think this might get fixed once we move error into core and turn its `Provider` impl into a blanket impl, if not I'll investigate alternative API structures that make this work in more places.

Reply:
I'd like to make sure that the ergonomics get a close look here before we stabilize, but this seems acceptable for now.

Reply:
I think you could use `Error::request_ref(&self.error)`. Maybe you need `<dyn Error>::request_ref(&self.error)` though. Both are nicer than this IMHO.

Reply:
The former won't work, but the latter will. While I agree I think my goal is still to be able to do self.error.request_ref() so I'd rather focus on the API rather than this specific callsite for now

Reply:
@yaahc Fair enough.